### ISSUE TITLE
Consider "recent sources" endpoint for high-volume instances

### ISSUE THREAD

**Issue by rocodes** [freedomofpress/securedrop#6539]
## Description

Organizations with very large numbers of sources are facing performance issues when using SecureDrop Workstation. One possibility is to create an endpoint that only returns, for example, sources `last_updated` in the last 24h. (The workstation would need to be configured to either sync to either "all sources" or "recent sources".  Further discussion of that issue and implementation details can take place in the workstation repo, if this is something we decide to pursue).
 
## User Research Evidence

Issue filed based on user feedback

## User Stories

As a frequent user of a high-volume SecureDrop, I want to be able to use the system, including SecureDrop Workstation.

(Discussion welcome please)

**Comment by zenmonkeykstop** [comment]
See also https://github.com/freedomofpress/securedrop-client/issues/1549 and #5104

**Comment by zenmonkeykstop** [comment]
FWIW I'm more in favour of a call like /sources/since/<timestamp> rather than one with a hardwired 24h window. This would also be more use for the syncing case, as syncs run pretty frequently so in most cases the .../since/ value would be a couple of minutes top and return an empty set.

**Comment by rocodes** [comment]
> FWIW I'm more in favour of a call like /sources/since/ rather than one with a hardwired 24h window. This would also be more use for the syncing case, as syncs run pretty frequently so in most cases the .../since/ value would be a couple of minutes top and return an empty set.

Yup, agreed- I'm favour of that as well, along with many of the other v2 changes.  In terms of quick action for news orgs in this position though, switching to a delta-sync instead of the current sync we have will take a fair bit of reworking in the client, so it would not be immediate. I think it's the right way to go long-term, but if we were thinking about a more time-sensitive solution, it would be something that keeps the sync logic on the client the same, just with a smaller subset of sources.

(I'm not saying we need to rush things - just laying out the pros and cons of each for discussion)

**Comment by nabla-c0d3** [comment]
For the API's design (and based on what I wrote at https://github.com/freedomofpress/securedrop-client/issues/1549#issuecomment-1229441745), I would recommend making this filter a query parameter for /sources instead of a subresource (ie. /sources/since). 

For example `/sources?filter_by_updated_since=<datetime>`

Using query parameters instead of subresources means that filters can easily be combined: `/sources?filter_by_updated_since=<datetime>&filter_by_some_other_thing=123` and also separates the filtering from the actual resources.

**Comment by eloquence** [comment]
Just noting here as well that current `last_updated` would likely to be insufficient for this as it only gets bumped on source submissions, not on journalist activity associated with a source (replies, starring, "seen" etc.).

**Comment by zenmonkeykstop** [comment]
Yup @eloquence, I think first order of business is figuring out if we can safely add a field like "last_interacted".

**Comment by cfm** [comment]
I'll take the liberty of closing this, since we went a different direction in #5104.
